flowermd/utils/utils.py

```diff
@@ -96,21 +96,26 @@ def _is_float(num):
         )
 
 
-def calculate_box_length(mass, density, fixed_L=None):
+def calculate_box_length(density, mass=None, n_beads=None, fixed_L=None):
```

Comment:
I wonder if it make things a little easier if we have two calculate box functions in our utils, one called `calculate_box_length_mass_density` and the other `calculate_box_length_number_density` (or better names). The reason I'm suggesting it is because having `mass` and `n_beads` parameter both in function interface makes it confusing for users, specially if we decide to remove `density` from the simulation's `run_update_volume`. Then these util functions are becoming interface functions, not just something we use internally in our codes, so better to keep them simple.

Reply:
That isn't a bad idea. I wonder for the sake of preventing code duplicaiton, we keep `calculate_box_length` as is, and make it a hidden function, then have separation forward facing functions that correctly pass in the mass and n_bead parameters. What do you think?

Reply:
Yeah that would work perfectly.

Reply:
Ok, now that I think about this some more, what if we do this:

1. Remove the `set_target_box` and `_calculate_L` methods in `System` since this is basically what we want the `utils` methods to do.
2. Create 2 `get_target_box` methods in `utils`. One for mass density and one for number density. These can use the same `_calculate_box_length` method in `utils`.
3. In `Pack` since we need an initial box to give mBuild/Packmol, we can check the units of `density` and call the corresponding `get_target_box_` method.

I think this will reduce code duplication and simplify the `Pack` class. The `target_box` attribute is really an artifact from polybinder where the attribute was called behind the scenes when shrinking, but this isn't how shrinking/updating volume works in flowerMD.

Reply:
That sounds reasonable. With this change, do we still want to keep `expand_factor` parameter in `pack`?

Reply:
I think so, we'll still call the `get_target_box` method automatically in `Pack`. The most recent push has the changes I suggested, I just need to finish updating any breaking unit tests now.